opal/mca/common/ucx/common_ucx_wpool.h

```diff
@@ -196,34 +215,35 @@ opal_common_ucx_tlocal_fetch(opal_common_ucx_wpmem_t *mem, int target,
                                 ucp_ep_h *_ep, ucp_rkey_h *_rkey,
                                 opal_common_ucx_winfo_t **_winfo)
 {
-    opal_common_ucx_tlocal_fast_ptrs_t *fp = NULL;
+    _mem_record_t *mem_rec = NULL;
     int expr;
     int rc = OPAL_SUCCESS;
 
     /* First check the fast-path */
-    rc = opal_tsd_tracked_key_get(&mem->mem_tls_key, (void**)&fp);
+    rc = opal_tsd_tracked_key_get(&mem->tls_key, (void**)&mem_rec);
     if (OPAL_SUCCESS != rc) {
         return rc;
     }
-    expr = fp && (NULL != fp->winfo) && (fp->winfo->endpoints[target]) &&
-            (NULL != fp->rkeys[target]);
+    expr = mem_rec && (NULL != mem_rec->winfo) && (mem_rec->winfo->endpoints[target]) &&
+            (NULL != mem_rec->rkeys[target]);
```

Comment:
Is it possible that mem_rec exists, but mem_rec->winfo is not set?
maybe it should go to become an assert
```suggestion
    is_ready = mem_rec && (mem_rec->winfo->endpoints[target]) &&
            (NULL != mem_rec->rkeys[target]);
    MCA_COMMON_UCX_ASSERT((NULL == mem_rec) || (NULL != mem_rec->winfo));
```